src/server/game/Entities/Creature/Creature.cpp

```diff
@@ -1018,8 +1018,7 @@ bool Creature::Create(ObjectGuid::LowType guidlow, Map* map, uint32 phaseMask, u
         TC_LOG_ERROR("entities.unit", "Creature::Create(): given coordinates for creature (guidlow %d, entry %d) are not valid (X: %f, Y: %f, Z: %f, O: %f)", guidlow, entry, pos.GetPositionX(), pos.GetPositionY(), pos.GetPositionZ(), pos.GetOrientation());
         return false;
     }
-    UpdatePositionData();
```

Comment:
This breaks SetZoneScript() (called in CreateFromProto). area/zone info must already be present there

Reply:
So we have a little problem here then. Object_field_scale must be initialized before calling for collision height which also happens in CreateFromProto otherwise the CollisionHeight call will crash on startup when creating battlefield npc's

Reply:
would that be a problem only with battlefield npcs ? maybe then that part has to be changed ?

Reply:
It has been changed already. Instead of using cached data the map helper is being used now.